src/network/networkd-address.c

```diff
@@ -448,6 +448,127 @@ static int address_set_masquerade(Address *address, bool add) {
+        if (r < 0)
+                return r;
+
+        if (address->family == AF_INET) {
+                r = sd_netlink_message_append_in_addr(m, NLBL_UNLABEL_A_IPV4ADDR, &address->in_addr.in);
+                if (r < 0)
+                        return r;
+
+                r = sd_netlink_message_append_in_addr(m, NLBL_UNLABEL_A_IPV4MASK, &mask.in);
+        } else if (address->family == AF_INET6) {
+                r = sd_netlink_message_append_in6_addr(m, NLBL_UNLABEL_A_IPV6ADDR, &address->in_addr.in6);
+                if (r < 0)
+                        return r;
+
+                r = sd_netlink_message_append_in6_addr(m, NLBL_UNLABEL_A_IPV6MASK, &mask.in6);
+        }
+        if (r < 0)
+                return r;
+
+        r = sd_netlink_send(genl, m, NULL);
+        return r < 0 ? r : 0;
+}
+
+static void address_add_netlabel(const Address *address) {
+        const Set *labels;
+        int r;
+        _cleanup_free_ char *addr_str = NULL;
+
+        assert(address);
+        assert(address->link);
+
+        if (!address->link->network)
+                return;
+
+        if (address->family == AF_INET)
+                labels = address->link->network->ipv4_netlabel_unlbl_labels;
+        else if (address->family == AF_INET6)
+                labels = address->link->network->ipv4_netlabel_unlbl_labels;
+        else
+                return;
+
+        int r2;
+
+        r2 = in_addr_to_string(address->family, &address->in_addr, &addr_str);
+
+        const char *label;
+        SET_FOREACH(label, labels) {
+                r = netlabel_op(true, label, address);
+                if (r < 0) {
+                        log_warning_errno(r, "Adding NetLabel %s for IP address %s/%d failed, ignoring: %m",
+                                          label,
+                                          r2 == 0? addr_str: "(error while translating to string)",
+                                          address->prefixlen);
+                }
+        }
+
+        log_debug("Added NetLabel %s for interface %s IP address %s/%d",
+                  label, address->link->ifname,
+                  r2 == 0? addr_str: "(error while translating to string)", address->prefixlen);
+}
+
+static void address_del_netlabel(const Address *address) {
+        int r;
+        _cleanup_free_ char *addr_str = NULL;
+
+        assert(address);
+        assert(address->link);
+
+        if (!address->link->network)
+                return;
+
+        int r2;
+        r2 = in_addr_to_string(address->family, &address->in_addr, &addr_str);
+
+        r = netlabel_op(false, NULL, address);
+        if (r < 0)
+                log_warning_errno(r, "Deleting NetLabel for IP address %s/%d failed, ignoring: %m",
+                                  r2 == 0? addr_str: "(error while translating to string)",
+                                  address->prefixlen);
+        log_debug("Deleted NetLabels for interface %s IP address %s/%d",
```

Comment:
Also here. Please use `log_link_debug()`, then `for interface %s` can be dropped.